src/main/scala/me/ex0ns/inlinexkcd/bot/InlineXKCDBot.scala

```diff
@@ -37,12 +39,15 @@ object InlineXKCDBot extends TelegramBot with Commands with Polling {
 
   def parseComic(notify : Boolean = false): Unit = {
 
-    def notifyAllGroups(url: String, alt: String) = {
+    def notifyAllGroups(url: String, title: String, text: String) = {
       Groups.all.map((documents) => {
         documents.grouped(MESSAGES_LIMIT).foreach((documents) => {
           documents.flatMap(_.get[BsonString]("_id")).map(_.getValue.toLong).foreach(id => {
+            api.request(SendMessage(Left(id), title, Some(ParseMode.Markdown)))
+            Thread.sleep(MESSAGE_ORDER_DELAY) // Avoid undeterministic sending order 
             api.request(SendPhoto(Left(id), Right(url)))
-            api.request(SendMessage(Left(id), alt))
+            Thread.sleep(MESSAGE_ORDER_DELAY)
```

Comment:
This will be fixed in the upcoming version of the library; using per-chat synchronous requests.